docs/guide/collections.md

```diff
@@ -0,0 +1,98 @@
+---
```

Comment:
tbh, I read this section multiple times and I'm still not sure what `with_collection` does

Reply:
Kind of meta, but I wonder how often collections are used with view components. Are they valuable? Do we use them? etc.

Reply:
I've never seen `with_collection` before reading this doc 😅

Reply:
https://github.com/github/view_component/pull/684/commits/0bc8acd195798f39b652bbb647773fbbbc00432d

Reply:
We have over a dozen uses internally. Not a ton, but it's definitely used 😄